electron/launcher/main-process/helpers/request-wrapper.js

```diff
@@ -0,0 +1,71 @@
+const rq = require('request-promise');
+const autoLogin = require('helpers/login').autoLogin;
+
+const config = require('config.json');
+const djangoUrl = config.DJANGO_SERVER.HOST +":" +config.DJANGO_SERVER.PORT;
+
+
+const checkSession = () => {
+  return new Promise((resolve, reject) => {
+    if(Date.now() > sessionExpiry){
+      resolve(false);
+    }
+    else{
+      resolve(true);
+    }
+  });
+}
+
+
+// THIS WRAPPERS INJECTS SENDS CSRF AND SESSION IN THE HEADER
+const request = options => {
+  return new Promise((resolve, reject) => {
+    // CHECK SESSION
+    checkSession().then(isFresh => {
+      if(isFresh){
+        return null;
+      }
+      else{
+        return autoLogin().then(response => {
+          return null;
+        })
+        .catch(error => {
+          throw error;
+        })
+      }
+    })
+    // MODIFY REQUEST HEADER
+    .then(val => {
+      // ONLY ADD CSRF AND SESSION TOKEN TO REQUESTS TO DJANGO SERVER
```

Comment:
similarly, `val` is always going to be null. Can we remove this variable?

Reply:
Removed